components/markdown_image.jsx

```diff
@@ -50,14 +54,47 @@ export default class MarkdownImage extends React.PureComponent {
         this.setState({loadFailed: true});
     }
 
+    isHeaderChangeMessage = () => {
+        return this.props.postType &&
+            this.props.postType === Constants.PostTypes.HEADER_CHANGE;
+    }
+
+    componentDidUpdate(prevProps) {
+        this.onUpdated(prevProps.src);
+    }
+
+    onUpdated = (prevSrc) => {
+        if (this.props.src && this.props.src !== prevSrc) {
+            this.setState({loadFailed: false});
+        }
+    }
+
+    handleImageLoaded = ({height, width}) => {
+        this.setState({
+            loaded: true,
+        }, () => { // Call onImageLoaded prop only after state has already been set
+            if (this.props.onImageLoaded) {
+                this.props.onImageLoaded({height, width});
+            }
+        });
+    }
+
     render() {
         const {imageMetadata, src, alt, imageIsLink} = this.props;
         if (src === '' || this.state.loadFailed) {
+            let className = 'markdown-inline-img broken-image';
+            if (this.isHeaderChangeMessage()) {
+                className += ' broken-image--scaled-down';
+            }
+
             return (
-                <img
-                    alt={alt}
-                    src={brokenImageIcon}
-                />
+                <div style={{display: 'inline-block'}}>
```

Comment:
I thought a class is not necessary for a one-off use case, but sure, can use a class instead

Reply:
Yes, please use a class.

Reply:
> Can we convert this to a class instead of using an inline style

Sorry, i am a little late to the comment here. It is fine to have simple styles in JS like this considering we have css which is not well maintained. If we make this a class we most likely will never get to cleaning this part if the html changes

Reply:
@devinbinnie, please confirm 🙂

Reply:
@abdusabri Sure that's fine